src/pytest_bdd/parser.py

```diff
@@ -453,9 +482,50 @@ def parse(self) -> Feature:
 
         for child in feature_data.children:
             if child.background:
-                feature.background = self.parse_background(child.background, feature)
+                feature.background = self.parse_background(child.background)
+            elif child.rule:
+                self._parse_and_add_rule(child.rule, feature)
             elif child.scenario:
-                scenario = self.parse_scenario(child.scenario, feature)
-                feature.scenarios[scenario.name] = scenario
+                self._parse_and_add_scenario(child.scenario, feature)
 
         return feature
+
+    def _parse_and_add_rule(self, rule_data: GherkinRule, feature: Feature) -> None:
+        """Parse a rule, including its background and scenarios, and add to the feature."""
+        rule = Rule(
+            keyword=rule_data.keyword,
+            name=rule_data.name,
+            description=rule_data.description,
+            tags=self.get_tag_names(rule_data.tags),
+        )
+
+        # Add background if present within the rule
+        background = self._extract_rule_background(rule_data)
+        if background:
+            rule.add_background(background)
```

Comment:
I guess we can avoid having this add_background method, and instead pass the background to the Rule constructor directly